### PR TITLE
PMM-7618 inventory and add instance migration

### PR DESCRIPTION
**Pull request by lunaticusgreen**
**What this PR does / why we need it**: Components migration to grafana

**Which issue(s) this PR fixes**: https://jira.percona.com/browse/PMM-7618, https://jira.percona.com/browse/PMM-7617

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -208,6 +208,8 @@
   },
   "dependencies": {
     "@grafana/slate-react": "0.22.9-grafana",
+    "history": "^4.10.1",
```
Comment (atymchuk) [inline package.json:211]: This needs to be sorted by alphabet

File: public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx [hunk 2]
```diff
@@ -18,7 +18,17 @@ export const DashboardLinksDashboard: React.FC<Props> = props => {
   const { link, linkInfo } = props;
   const listRef = useRef<HTMLUListElement>(null);
   const [opened, setOpened] = useState(0);
-  const resolvedLinks = useResolvedLinks(props, opened);
+  let resolvedLinks = useResolvedLinks(props, opened);
+
+  // TODO: remove it ASAP after migration transition period is finished
```
Comment (atymchuk) [inline public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx:23]: Better drop a ticket, or it's easy to forget

File: public/app/percona/check/Check.service.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
+// @ts-nocheck
```
Comment (atymchuk) [inline public/app/percona/check/Check.service.ts:1]: Is it that bad that it needs `nocheck`?

File: public/app/percona/check/CheckPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
+import React, { FC, useEffect, useMemo, useState } from 'react';
+// @ts-ignore
```
Comment (atymchuk) [inline public/app/percona/check/CheckPanel.tsx:2]: Is there no other way, other than disabling ts? I suggest we create a tech-debt ticket to fix ts everywhere.

File: public/app/percona/inventory/ColumnRenderers/ColumnRenderers.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+export const agentsDetailsRender = (element: any) => {
+  const mainColumns = ['agent_id', 'type', 'isDeleted', 'service_ids', 'custom_labels'];
+  const labels = Object.keys(element).filter(label => !mainColumns.includes(label));
+
+  return (
+    <div className={styles.detailsWrapper}>
+      {labels.map((label, key) => (element[label] ? <span key={key}>{`${label}: ${element[label]}`}</span> : null))}
+      {element.username ? <span>password: ******</span> : null}
+      {element.service_ids && element.service_ids.length ? (
+        /* TODO(lunaticusgreen): all map renders need a key! */
```
Comment (atymchuk) [inline public/app/percona/inventory/ColumnRenderers/ColumnRenderers.tsx:28]: I remember tslint was yelling at this :)

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Comment by atymchuk** [comment]
@lunaticusgreen I understand the volume, but pls see what we can do to address disabled ts checks.

**Comment by lunaticusgreen** [comment]
@atymchuk Thanks, very good point! I'll either fix everything you mentioned or create a ticket for every problem we'll decide to fix later

File: package.json (hunk 1 above)
Comment (lunaticusgreen) [reply package.json]: Done

File: public/app/percona/check/Check.service.ts (hunk 3 above)
Comment (lunaticusgreen) [reply public/app/percona/check/Check.service.ts]: Thanks, checked and fixed TS, nocheck removed!

File: public/app/percona/check/CheckPanel.tsx (hunk 4 above)
Comment (lunaticusgreen) [reply public/app/percona/check/CheckPanel.tsx]: Got rid of history and router here completely

File: public/app/percona/inventory/ColumnRenderers/ColumnRenderers.tsx (hunk 5 above)
Comment (lunaticusgreen) [reply public/app/percona/inventory/ColumnRenderers/ColumnRenderers.tsx]: Fixed

File: public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx (hunk 2 above)
Comment (lunaticusgreen) [reply public/app/features/dashboard/components/SubMenu/DashboardLinksDashboard.tsx]: Added ticket PMM-7736, thanks

**Review by atymchuk — APPROVED** [review]
Good job, thanks!

**Pull request merged**